tests/README.md

```diff
@@ -130,16 +128,116 @@ To use Docker backend, you need to have:
 
 The procedure is same as using Podman, you just swap the `podman` call with `docker`. But since Docker does not support rootless containers you will have to take superuser route of the guide.
 
-# How to run the tests
 
-To test you profile or rule use `test_suite.py` script. It can take your DataStream, and test it on the specified backend.
+# Test scenarios
+
+The test scenarios are used to test rules. They modify the system configuration
+so that OpenSCAP can return expected results.
+
+The test scenarios for a rule are located in `tests` subdirectory in rule
+directory. The test scenarios are written in Bash. The format of the file name
+is `scenario_name.expected_scan_result.sh`, where `scenario_name` is an
+arbitrary name, and `expected_scan_result` is the result of the evaluation of
+the rule that `oscap` should return when the rule is evaluated.
+`expected_scan_result` can be one of `pass`, `fail`  or `notapplicable`. It's
+very important to keep this naming form.
+
+For example:
+* `something.pass.sh`: Success scenario - script is expected to prepare machine
+  in such way that the rule is expected to pass.
+* `something.fail.sh`: Fail scenario - script is expected to break machine so
+  the rule fails. Test Suite than, if initial scan fails as expected, tries to
+  remediate machine and expects evaluation to pass after the remediation.
+
+## Scenarios format
+
+Scenarios are simple Bash scripts. A scenario starts with a header which
+provides metadata. The metadata are parsed by the test framework and affect the
+test runs. After the header, arbitrary Bash commands can follow.
+
+The header consists of comments (starting by `#`). Possible keys are:
+
+- `packages` is a comma-separated list of packages to install.
+- `platform` is a comma-separated list of platforms where the test scenario can
+  be run. This is similar to `platform` used in our remediations. Examples of
+  values: `multi_platform_rhel`, `Red Hat Enterprise Linux 7`,
+  `multi_platform_all`. If `platform` is not specified in the header,
+  `multi_platform_all` is assumed.
+- `profiles` is a comma-separated list of profiles to which this scenario is
+  restricted. Use this only if the scenario makes sense only in a specific
+  profile. Typically, a rule doesn't depend on a profile and behaves the same
+  way regardless the profile it's a part of. If the rule is parametrized by
+  variables (XCCDF Values), use the `variables` key instead. This key is
+  intended to be used in regression testing of bugs in profiles, it isn't
+  intended for casual use.
+- `remediation` is a string specifying one of the allowed remediation types (eg.
+  `bash`, `ansible`, `none`). The `none` value means that the tested rule has no
+  implemented remediation.
```

Comment:
We can add here that `none` can also be used in case that remediation breaks test environemnt (for example unmounting `/tmp` would break test suite as opnescap generates reports into this directory).